### PR TITLE
Gunslinger Attacks After Aiming

### PR DESCRIPTION
**Pull request by Anisetobegan**

### CODE REVIEW

File: Assets/Scripts/GunSlingerScript.cs [hunk 1]
```diff
@@ -1,39 +1,216 @@
 using System.Collections;
 using System.Collections.Generic;
+using Unity.VisualScripting;
 using UnityEngine;
+using static UnityEngine.GraphicsBuffer;
```
Comment (morathias) [inline Assets/Scripts/GunSlingerScript.cs]: Remove unnecessary usings

File: Assets/Scripts/GunSlingerScript.cs [hunk 2]
```diff
@@ -1,39 +1,216 @@
...
+                if (enumerator == null)
+                {
+                    Attack();
+                }
+
+                break;
+
+            case State.Reloading:
+
+                Reloading();
```
Comment (morathias) [inline Assets/Scripts/GunSlingerScript.cs]: check if enumerator is running here too

**Review by morathias — CHANGES_REQUESTED** [review]

File: Assets/Scripts/GunSlingerScript.cs (hunk 1 above)
Comment (Anisetobegan) [reply Assets/Scripts/GunSlingerScript.cs]: Done

File: Assets/Scripts/GunSlingerScript.cs (hunk 2 above)
Comment (Anisetobegan) [reply Assets/Scripts/GunSlingerScript.cs]: Done

**Review by morathias — APPROVED** [review]

**Pull request merged**